utils/compile-time-utils/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+pub use git_version::git_version;
+
+#[macro_export]
+macro_rules! crate_version_commit {
+    () => {
+        Box::leak(
+            [
+                env!("CARGO_PKG_VERSION"),
+                "-",
+                ya_compile_time_utils::git_version!(),
```

Comment:
(not tested)

This should strip the `-` character if it's not possible to retrieve the git version (and thus does not require git since it would fail silently):
`git_version!(prefix = "-", fallback = "")`